src/modules/flyout/flyout-adapter.service.ts

```diff
@@ -1,61 +1,40 @@
 import {
-  ApplicationRef,
-  ComponentFactoryResolver,
-  ComponentRef,
-  EmbeddedViewRef,
+  ElementRef,
   Injectable,
-  Injector,
   Renderer2,
   RendererFactory2
 } from '@angular/core';
 
-import { SkyFlyoutComponent } from './index';
 import { SkyWindowRefService } from '../window';
 
 @Injectable()
 export class SkyFlyoutAdapterService {
-
   private renderer: Renderer2;
 
   constructor(
-    private appRef: ApplicationRef,
-    private injector: Injector,
     private rendererFactory: RendererFactory2,
-    private resolver: ComponentFactoryResolver,
     private windowRef: SkyWindowRefService
   ) {
-      this.renderer = this.rendererFactory.createRenderer(this.windowRef.getWindow().document.body, undefined);
+    this.renderer = this.rendererFactory.createRenderer(undefined, undefined);
   }
 
-  public appendHostEl(): ComponentRef<SkyFlyoutComponent> {
-    const componentRef = this.resolver
-      .resolveComponentFactory(SkyFlyoutComponent)
-      .create(this.injector);
-
-    const domElem = (componentRef.hostView as EmbeddedViewRef<any>)
-      .rootNodes[0] as HTMLElement;
-
-    const windowObj = this.windowRef.getWindow();
-    this.renderer.appendChild(windowObj.document.body, domElem);
-
-    return componentRef;
+  public appendToBody(element: any): void {
+    const body = this.windowRef.getWindow().document.body;
+    this.renderer.appendChild(body, element);
   }
 
-  public removeHostEl(componentRef: ComponentRef<SkyFlyoutComponent>): void {
-    this.appRef.detachView(componentRef.hostView);
-    componentRef.destroy();
+  public removeHostElement(): void {
+    const document = this.windowRef.getWindow().document;
+    const hostElement = document.querySelector('sky-flyout');
+    this.renderer.removeChild(document.body, hostElement);
   }
 
-  public getCurrentActiveElement(): HTMLElement {
-    const docObj = this.windowRef.getWindow().document;
-    return <HTMLElement> docObj.activeElement;
-  }
+  public adjustHeaderForHelp(header: ElementRef): void {
+    const windowObj = this.windowRef.getWindow();
+    const helpWidget = windowObj.document.getElementById('bb-help-invoker');
```

Comment:
Just curious - why use `.getElementById()` over other selectors? @Blackbaud-SteveBrush

Reply:
There's a significant performance boost when using the more specific `getElementById` method: https://jsperf.com/getelementbyid-vs-queryselector